smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -417,7 +418,7 @@ private void generateClient(ServiceShape shape) {
         // Generate the non-modular service client.
         Symbol serviceSymbol = symbolProvider.toSymbol(shape);
         String nonModularName = serviceSymbol.getName().replace("Client", "");
-        String filename = serviceSymbol.getDefinitionFile().replace("Client", "");
+        String filename = CodegenUtils.SOURCE_FOLDER + "/" + serviceSymbol.getDefinitionFile().replace("Client", "");
```

Comment:
You should update the definition file on the service symbol instead of prefixing here.

Reply:
Done in [`f63e4ca` (#432)](https://github.com/awslabs/smithy-typescript/pull/432/commits/f63e4cab1652c8f3eec6be6413ceea493a762e40)